nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryQuery.java

```diff
@@ -2533,6 +2533,26 @@ public void testLTTimeWithMilliseconds() {
         assertEquals(1, docs.size());
     }
 
+    @Test
+    public void testQueryNow() throws Exception {
+        setupDocTest();
+        // sleep a second to be sure that the document is found
```

Comment:
You can use  waitForFulltextIndexing(); instead of nextTx + sleep

Reply:
But `waitForFulltextIndexing` has different semantics. Here the sleep is to wait for time to advance.